### PR TITLE
feat: add export genesis script

### PR DESCRIPTION
**Pull request by peterwht**

### CODE REVIEW

File: scripts/export-genesis-state.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# Check if the chain spec location exists
+if [ ! -f "$chain_spec_location" ]; then
+    echo "Error: Chain spec location '$chain_spec_location' does not exist."
+    exit 1
+fi
+
+# Export the genesis state
+export_output="$("$binary_location" export-genesis-state --chain="$chain_spec_location" "$output_file_name" 2>&1 | tee /dev/tty)"
+
+echo ""
```
Comment (Moliholy) [inline scripts/export-genesis-state.sh:28]: What's this for? This dangerously looses the returned code of the previous command, i.e. `$?`.

File: scripts/export-genesis-state.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+# Check if the export command produced an error
+if [[ "$export_output" == *"No specific runtime was recognized for ChainSpec's Id"* || "$export_output" == *"Error"* ]]; then
```
Comment (Moliholy) [inline scripts/export-genesis-state.sh:31]: Why not simply checking the returned code?

```sh
"$binary_location" export-genesis-state --chain="$chain_spec_location" "$output_file_name"

if [ $? -ne 0 ]; then
    echo "Error exporting genesis state. Please check your chain spec file."
    echo "Deleting '$output_file_name'..."
    rm "$output_file_name"
    exit 1
else
    echo "Genesis state exported successfully to '$output_file_name'"
fi

File: scripts/export-genesis-state.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+fi
+
+# Check if the chain spec location exists
+if [ ! -f "$chain_spec_location" ]; then
+    echo "Error: Chain spec location '$chain_spec_location' does not exist."
+    exit 1
+fi
+
+# Export the genesis state
+export_output="$("$binary_location" export-genesis-state --chain="$chain_spec_location" "$output_file_name" 2>&1 | tee /dev/tty)"
```
Comment (Moliholy) [inline scripts/export-genesis-state.sh:26]: What's the `tree` for? It seems it simply duplicates the input to another terminal, isn't it? And if so, is there any special reason?

File: scripts/export-genesis-state.sh [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+echo ""
+
+# Check if the export command produced an error
+if [[ "$export_output" == *"No specific runtime was recognized for ChainSpec's Id"* || "$export_output" == *"Error"* ]]; then
+    echo "Error exporting genesis state. Please check your chain spec file."
+    echo "Deleting '$output_file_name'..."
+    rm "$output_file_name"
```
Comment (Moliholy) [inline scripts/export-genesis-state.sh:34]: Without the `-f` it's going to ask for confirmation. We should hence add it.

**Review by Moliholy — CHANGES_REQUESTED** [review]

File: scripts/export-genesis-state.sh (hunk 3 above)
Comment (peterwht) [reply scripts/export-genesis-state.sh:26]: If `2>&1` is removed to output to the console, then `export_output` remains empty. The `tee` allows for console output while setting `export_output` still.

File: scripts/export-genesis-state.sh (hunk 4 above)
Comment (peterwht) [reply scripts/export-genesis-state.sh:34]: Good idea

**Review by Moliholy — APPROVED** [review]

**Pull request merged**